src/lld/moviebooking/class_diagram.puml

```diff
@@ -1,3 +1,323 @@
+    interface TheatreMovieReadDao{
+        + findAllTheatres(movieId: long, cityId: long): List<TheatreShowDTO>
+        + findAllTheatres(cityId: long): List<TheatreShowDTO>
+        + findAllMovies(cityId: long): List<Movie>
+        + findAllMovies(theatreId: long, cityId: long): List<Movie>
+    }
+    interface TheatreMovieWriteDao{
+        + addMovieTheatreCity(cityId: long, movieId: long, theatreId: long): boolean
+    }
+    class DBTheatreMovieReadDao implements TheatreMovieReadDao {
+        - connection: Connection
+        + DBTheatreMovieReadDao()
+        + findAllTheatres(movieId: long, cityId: long): List<TheatreShowDTO>
+        + findAllTheatres(cityId: long): List<TheatreShowDTO>
+        + findAllMovies(cityId: long): List<Movie>
+        + findAllMovies(theatreId: long, cityId: long): List<Movie>
+    }
+    class DBTheatreMovieWriteDao implements TheatreMovieWriteDao {
+        - connection: Connection
+        + DBTheatreMovieWriteDao()
+        + addMovieTheatreCity(cityId: long, movieId: long, theatreId: long): boolean
+    }
+'    FR3: show all seats for the show, update/add show seat
+    interface ShowSeatReadDao{
+        + findAllSeats(showId: long): List<SeatDTO>
+    }
+    interface ShowSeatWriteDao {
+        + addShowSeat(showId: long, seatId: long, seatStatusId: int, seatCategoryId: int): SeatDTO
+        + updateShowSeat(showId: long, seatId: long, seatStatusId: int, seatCategoryId: int): SeatDTO
+    }
+    class DBShowSeatReadDao implements ShowSeatReadDao{
+        - connection: Connection
+        + DBShowSeatReadDao()
+        + findAllSeats(showId: long): List<SeatDTO>
+    }
+    class DBShowSeatWriteDao implements ShowSeatWriteDao{
+        - connection: Connection
+        + DBShowSeatWriteDao()
+        + addShowSeat(showId: long, seatId: long, seatStatusId: int, seatCategoryId: int): SeatDTO
+        + updateShowSeat(showId: long, seatId: long, seatStatusId: int, seatCategoryId: int): SeatDTO
+    }
+    'FR4: book ticket after selecting seat
+    interface BookSeatReadDao{
+        + findBooking(booking: id): Booking
+        + findAllBookings(): List<Booking>
+        + findAllBookings(userId: long): List<Booking>
+    }
+    interface BookSeatWriteDao{
+        + createBooking(booking: Booking)
+        + updateBooking(bookingId: long, booking: Booking)
+    }
+    class DBBookSeatReadDao implements BookSeatReadDao{
+        - connection: Connection
+        + DBBookSeatReadDao()
+        + findBooking(booking: id): Booking
+        + findAllBookings(): List<Booking>
+        + findAllBookings(userId: long): List<Booking>
+    }
+    class DBBookSeatWriteDao implements BookSeatWriteDao{
+        - connection: connection
+        + DBBookSeatWriteDao()
+        + createBooking(booking: Booking)
+        + updateBooking(bookingId: long, booking: Booking)
+    }
+}
+
+'Relationship
+'Movie --> MovieReleaseDetails : has
+Theatre --> Screen : has
+Theatre --> Address : has
+Screen --> Seat
```

Comment:
It is better to write service class definition to understand how api is implemented. Dao, models will only reflect part of the system, To display behavior of complete system, we would need service and probably our api definitions